microsoft/vscode-dev-containers
# Onboard Jekyll to image build

This implements #919 to generate a jekyll image. Like other images, it will include nvm and yarn, but does not include a specific version of Node.js in `base.Dockerfile`.  Instead this is added in the user `Dockerfile` based on a version in `devcontainer.json`. The user will be prompted for which version when this definition is added via **Add Development Configuration Files**.  Unneeded files (like `base.Dockerfile`) are also filtered out in this case.

To test the image alone:

1. Open this repository in a Codespace
2. Run the following in a terminal in the repository root
    ```bash
    yarn install
    build/vscdc push --no-push --registry mcr.microsoft.com --registry-path vscode/devcontainers --release main jekyll
    docker run -it -u vscode --rm mcr.microsoft.com/vscode/devcontainers/jekyll:dev bash
    ```
3. Once in the container, run `nvm install lts/*` to ensure node installs
4. Do any verification you want to do.

Once merged, `mcr.microsoft.com/vscode/devcontainers/jekyll:dev` will be published, and on release the inital version of the image will go (currently set to `0.0.1`).
